pallet/staking/src/lib.rs

```diff
@@ -62,120 +68,26 @@ use sp_runtime::{
-
-	/// Add stakes to the staking pool.
-	///
-	/// This will transfer the stakes to a pallet/contact account.
-	fn stake(who: &Self::AccountId, item: Self::Item) -> DispatchResult;
-
-	/// Withdraw stakes from the staking pool.
-	///
-	/// This will transfer the stakes back to the staker's account.
-	fn unstake(who: &Self::AccountId, item: Self::Item) -> DispatchResult;
-}
-/// Extended stake trait.
-///
-/// Provide a way to access the deposit RING amount.
-pub trait StakeExt: Stake {
-	/// Amount type.
-	type Amount;
-
-	/// Get the staked amount.
-	fn amount(who: &Self::AccountId, item: Self::Item) -> Result<Self::Amount, DispatchError>;
-}
-
-/// A convertor from collators id. Since this pallet does not have stash/controller, this is
-/// just identity.
-pub struct IdentityCollator;
-impl<T> Convert<T, Option<T>> for IdentityCollator {
-	fn convert(t: T) -> Option<T> {
-		Some(t)
-	}
-}
-
-/// Staking ledger.
-#[derive(PartialEqNoBound, EqNoBound, Encode, Decode, MaxEncodedLen, TypeInfo, RuntimeDebug)]
-#[scale_info(skip_type_params(T))]
-pub struct Ledger<T>
-where
-	T: Config,
-{
-	/// Staked RING.
-	pub staked_ring: Balance,
-	/// Staked KTON.
-	pub staked_kton: Balance,
-	/// Staked deposits.
-	pub staked_deposits: BoundedVec<DepositId<T>, T::MaxDeposits>,
-	/// The RING in unstaking process.
-	pub unstaking_ring: BoundedVec<(Balance, T::BlockNumber), T::MaxUnstakings>,
-	/// The KTON in unstaking process.
-	pub unstaking_kton: BoundedVec<(Balance, T::BlockNumber), T::MaxUnstakings>,
-	/// The deposit in unstaking process.
-	pub unstaking_deposits: BoundedVec<(DepositId<T>, T::BlockNumber), T::MaxUnstakings>,
-}
-impl<T> Ledger<T>
-where
-	T: Config,
-{
-	fn is_empty(&self) -> bool {
-		self.staked_ring == 0
-			&& self.staked_kton == 0
-			&& self.staked_deposits.is_empty()
-			&& self.unstaking_ring.is_empty()
-			&& self.unstaking_kton.is_empty()
-			&& self.unstaking_deposits.is_empty()
-	}
-}
-
-/// A snapshot of the stake backing a single collator in the system.
-#[derive(Encode, Decode, MaxEncodedLen, TypeInfo, RuntimeDebug)]
-pub struct Exposure<AccountId> {
-	/// The total power backing this collator.
-	pub total: Power,
-	/// Nominators' stake power.
-	pub nominators: Vec<IndividualExposure<AccountId>>,
-}
-/// A snapshot of the staker's state.
-#[derive(Encode, Decode, MaxEncodedLen, TypeInfo, RuntimeDebug)]
-pub struct IndividualExposure<AccountId> {
-	/// Nominator.
-	pub who: AccountId,
-	/// Nominator's stake power.
-	pub value: Power,
-}
-
 #[frame_support::pallet]
 pub mod pallet {
 	// darwinia
 	use crate::*;
 
+	// Deposit helper for runtime benchmark.
+	#[cfg(feature = "runtime-benchmarks")]
+	use darwinia_deposit::Config as DepositConfig;
+	/// Empty trait acts as a place holder to satisfy the `#[pallet::config]` macro.
+	#[cfg(not(feature = "runtime-benchmarks"))]
+	pub trait DepositConfig {}
+
 	#[pallet::config]
-	pub trait Config: frame_system::Config {
+	pub trait Config: frame_system::Config + DepositConfig {
```

Comment:
Do we still need the `DepositConfig` bound after removing the `KtonHelper`?

Reply:
Yep. Because of `<T as deposit::Config>::Kton`.

Reply:
Also
```rs
fn deposit_for<T>(who: &T::AccountId, count: u32) -> Vec<DepositId<T>>
	where
		T: Config + darwinia_deposit::Config,
{
	(0..count.min(<<T as Config>::MaxDeposits>::get()) as u8)
		.map(|x| {
			<darwinia_deposit::Pallet<T>>::lock(
				RawOrigin::Signed(who.to_owned()).into(),
				UNIT,
				1,
			)
			.unwrap();

			x.into()
		})
		.collect()
}
```